repos/system_upgrade/el7toel8/actors/postgresqlcheck/libraries/library.py

```diff
@@ -0,0 +1,78 @@
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+
+def get_postgresql_packages():
+    """ Get list of installed PostgreSQL packages """
+
+    for rpm_pkgs in api.consume(InstalledRedHatSignedRPM):
+        return [pkg for pkg in rpm_pkgs.items if pkg.name.startswith('postgresql')]
+
+def get_postgresql_extensions():
+    """ Get list of PostgreSQL extensions """
+    result = []
+    try:
+        output = run([
+            '/usr/bin/su',
+            '-l', 'postgresql'
+            '-c', 'psql -tAc \'\dx\''
+        ], split=True)['stdout']
+    except CalledProcessError:
+        api.current_logger().debug('Unable to list PostgreSQL extensions: psql failed to run')
+        return False
+
+    for entry in output:
+        entry = entry.strip()
+        if not entry:
+            continue
+        name, version, schema, description = entry.split('|')
+        result.append({'name': name,
+                       'version': version,
+                       'schema': schema,
+                       'description': description})
+
+    return result
+
+def generate_report(packages):
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = 'https://www.postgresql.org/docs/10/upgrading.html'
+    MISSING_EXTENSIONS = ['dummy_seclabel', 'test_parser', 'tsearch2']
+
+    extensions = get_postgresql_extensions()
+
+    for pkg in packages:
+        # Show documentation url if postgresql-server installed
+        if pkg.name == 'postgresql-server':
+            title = 'PostgreSQL server installation detected'
+            severity = 'high'
+            summary = 'PostgreSQL server will be upgraded. Additional steps might be required.\n\n' \
```

Comment:
I'd be be more concrete here. For example:
"PostgreSQL server component will be upgraded. Since RHEL-8 includes PostgreSQL server 10 by default, which is incompatible with 9.2 included in RHEL-7, it is necessary to proceed with additional steps for complete upgrade of the PostgreSQL data. Back-up your data before proceeding with the upgrade and read more about upgrade steps at <KB_article_url>."

Reply:
I've used text you've provided here (thank you), and included a KB article URL. That article is empty ATM, but will be filled in as soon as I get proper permissions. This should be sufficient for the sake of merging this PR.